specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json

```diff
@@ -1,745 +1,886 @@
+                }
+            ],
+            "properties": {
+                "durationOfCall": {
+                    "description": "The duration of the call",
+                    "type": "number",
+                    "format": "double"
+                }
+            }
+        },
+        "AcsCallParticipantAddedEventData": {
+            "description": "Schema of the Data property of an EventGridEvent for a Microsoft.Communication.CallParticipantAdded event.",
+            "type": "object",
+            "allOf": [
+                {
+                    "$ref": "#/definitions/AcsCallingParticipantEventBaseProperties"
+                }
+            ]
+        },
+        "AcsCallParticipantRemovedEventData": {
+            "description": "Schema of the Data property of an EventGridEvent for a Microsoft.Communication.CallParticipantRemoved event.",
+            "type": "object",
+            "allOf": [
+                {
+                    "$ref": "#/definitions/AcsCallingParticipantEventBaseProperties"
+                }
+            ]
+        },
+        "AcsCallingParticipantEventBaseProperties": {
+            "description": "Schema of common properties of all Calling Participant events",
+            "type": "object",
+            "allOf": [
+                {
+                    "$ref": "#/definitions/AcsCallingEventBaseProperties"
+                }
+            ],
+            "properties": {
+                "userCommunicationIdentifier": {
+                    "description": "The communication identifier of the user who was added to the call",
+                    "$ref": "../../../../../communication/data-plane/Common/stable/2021-03-07/common.json#/definitions/CommunicationIdentifierModel"
+                },
+                "participantId": {
+                    "description": "The id of the user who was added",
+                    "type": "string"
+                },
+                "displayName": {
+                    "description": "The display name of the user who was added",
+                    "type": "string"
+                },
+                "endpointType": {
+                    "description": "The type of the user's endpoint added",
+                    "type": "string"
+                }
+            }
+        },
+        "AcsCallingEventBaseProperties": {
+            "description": "Schema of common properties of all calling events",
+            "type": "object",
+            "properties": {
+                "startedByCommunicationIdentifier": {
+                    "description": "The communication identifier of the user who started the call",
+                    "$ref": "../../../../../communication/data-plane/Common/stable/2021-03-07/common.json#/definitions/CommunicationIdentifierModel"
+                },
+                "serverCallId": {
+                    "description": "The id of the server call",
+                    "type": "string"
+                },
+                "isTwoParty": {
```

Comment:
Should these properties be marked as required?

Reply:
not sure, is that a new requirement?

Reply:
Will the service always populate all of these properties or will some of them be left blank?

This impacts how the properties are generated in the C# SDK - non-reference type properties that are not marked as required will be generated as nullable.